### PR TITLE
Added neighbor_or_periodic_neighbor

### PR DESCRIPTION
**Pull request by samiiali**
Hopefully this should do it.

### CODE REVIEW

File: include/deal.II/grid/tria_accessor.h [hunk 1]
```diff
@@ -2527,6 +2527,16 @@ class CellAccessor :  public TriaAccessor<dim,dim,spacedim>
   periodic_neighbor (const unsigned int i) const;
 
   /**
+   * For a cell whose @c ith face is not at a boundary, this function returns
+   * the same result as neighbor(). If the @c ith face is at a periodic boudnary
```
Comment (masterleinad) [inline include/deal.II/grid/tria_accessor.h]: boudnary -> boundary

File: include/deal.II/grid/tria_accessor.h [hunk 2]
```diff
@@ -2527,6 +2527,16 @@ class CellAccessor :  public TriaAccessor<dim,dim,spacedim>
...
+   * this function returns the same result as periodic_neighbor(). If neither of
+   * the afformentioned conditions are met, i.e. the @c ith face is on a
```
Comment (masterleinad) [inline include/deal.II/grid/tria_accessor.h]: afformentioned -> aforementioned

**Comment by samiiali** [comment]
By the way, when I am trying to compile this branch I get an error, that I was not getting before: 
```
error: prototype for ‘void dealii::LinearAlgebra::ReadWriteVector<Number>::import(const dealii::PETScWrappers::MPI::Vector&, dealii::VectorOperation::values, const dealii::LinearAlgebra::CommunicationPatternBase*)’ does not match any in class ‘dealii::LinearAlgebra::ReadWriteVector<Number>’
   ReadWriteVector<Number>::import(const PETScWrappers::MPI::Vector &petsc_vec,
```

**Comment by masterleinad** [comment]
This is not related to our commits...

**Comment by masterleinad** [comment]
The PR looks good.

**Pull request merged**